presto-main/src/main/java/com/facebook/presto/operator/scalar/QuantileDigestFunctions.java

```diff
@@ -101,6 +127,49 @@ public static Block valuesAtQuantilesBigint(@SqlType("qdigest(bigint)") Slice in
         return output.build();
     }
 
+    @SqlNullable
+    @ScalarFunction("truncated_mean")
+    @Description("The approx arithmetic mean, estimated from qdigest, of values in the quantile range bounded by lowerQuantile and upperQuantile.")
+    @SqlType(StandardTypes.DOUBLE)
+    public static Double truncatedMeanDouble(@SqlType("qdigest(double)") Slice input,
+            @SqlType(StandardTypes.DOUBLE) double lowerQuantile, @SqlType(StandardTypes.DOUBLE) double upperQuantile)
+    {
+        validateQuantileRange(lowerQuantile, upperQuantile);
+        QuantileDigest digest = new QuantileDigest(input);
+        List<Long> bounds = digest.getQuantiles(ImmutableList.of(lowerQuantile, upperQuantile));
+        return digestToTruncatedMean(digest, bounds, DOUBLE_MIDDLE_FUNCTION);
+    }
+
+    @SqlNullable
+    @ScalarFunction("truncated_mean")
+    @Description("The approx arithmetic mean, estimated from qdigest, of values in the quantile range bounded by lowerQuantile and upperQuantile.")
+    @SqlType(StandardTypes.REAL)
+    public static Long truncatedMeanReal(@SqlType("qdigest(real)") Slice input,
```

Comment:
For multiple parameters which are too long when put on a single line we tend to do put each on a separate line.

```java
public static Long truncatedMeanReal(
    @SqlType("qdigest(real)") Slice input,
    @SqlType(StandardTypes.DOUBLE) double lowerQuantile,
    @SqlType(StandardTypes.DOUBLE) double upperQuantile)
{```